onnxscript/function_libs/torch_lib/new_graph_prototype.py

```diff
@@ -0,0 +1,626 @@
+# -------------------------------------------------------------------------
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License.
+# --------------------------------------------------------------------------
+
+from __future__ import annotations
+
+from typing import Any, Dict, List, Mapping, Optional, Sequence, Tuple, Union
+
+import numpy as np
+import onnx
+import onnx.checker
+import onnx.defs
+import onnx.helper
+import onnx.shape_inference
+import torch
+from typing_extensions import TypeAlias
+
+import onnxscript
+from onnxscript import evaluator, onnx_opset
+from onnxscript import tensor as onnxscript_tensor
+from onnxscript._internal import param_manipulation, runtime_typing
+
+__all__ = [
+    "Graph",
+    "GraphEvaluator",
+    "GraphTensor",
+]
+
+
+ValidArgumentType: TypeAlias = Union[
+    "GraphTensor",
+    Sequence["GraphTensor"],
+    Sequence[float],
+    Sequence[int],
+    str,
+    int,
+    float,
+    bool,
+    None,
+]
+
+
+class GraphTensor(onnxscript_tensor.Tensor):
+    """An implementation of ONNX Tensors, based on a wrapper around numpy arrays.
```

Comment:
Is this meant to be a wrapper around torch.Tensor?

Reply:
answer: not really. a "constant value" (represented by a torch.Tensor or numpy.array) is a special case of the more general "value" class that this corresponds to.

Reply:
Specifically, inheriting from onnxscript's Tensor doesn't sound right here.